tools/bpf2c/bpf_code_generator.cpp

```diff
@@ -473,7 +474,11 @@ bpf_code_generator::encode_instructions()
                 output.lines.push_back(format_string("%s = %s;", destination, source));
             } else {
                 std::string source;
-                source = format_string("_maps[%s].address", std::to_string(map_definitions[output.relocation].index));
+                auto map_definition = map_definitions.find(output.relocation);
+                if (map_definition == map_definitions.end()) {
+                    throw std::runtime_error(std::string("Map ") + output.relocation + std::string(" doesn't exist"));
+                }
+                source = format_string("_maps[%s].address", std::to_string(map_definition->second.index));
                 output.lines.push_back(format_string("%s = POINTER(%s);", destination, source));
             }
         } break;
```

Comment:
(Not relevant to these set of changes) consider having a different function for each case. Otherwise the switch statement grows too long.